sdk/historian/src/main/java/org/dsa/iot/historian/database/Watch.java

```diff
@@ -28,25 +39,27 @@
     private Node realTimeValue;
     private String watchedPath;
 
-    private Node lastWrittenValue;
     private Node startDate;
     private Node endDate;
     private boolean enabled;
 
     // Values that must be handled before the buffer queue
     private long lastWrittenTime;
+
+    // Used to represent last value to the database
+    private Node lastWrittenValue;
+
+    // Used for POINT_CHANGE
     private Value lastValue;
 
     public WatchUpdate getLastWatchUpdate() {
-        if (lastWatchUpdate == null) {
```

Comment:
That's basically that. However, I haven't tested changing the value in-between. Could be a great test to write, however it's a bit hard, since this error didn't happen consistant way.